dags/s3_xml_import_pipeline.py

```diff
@@ -0,0 +1,245 @@
+import json
+import os
+from datetime import timedelta
+from airflow import DAG
+from airflow.models import Variable
+from airflow.models.dagrun import DagRun
+from airflow.operators.python_operator import ShortCircuitOperator
+
+from data_pipeline.dag_pipeline_config.xml_config import eJPXmlDataConfig
+from data_pipeline.etl_state import get_stored_state
+from data_pipeline.etl import (
+    etl_ejp_xml_zip,
+)
+from data_pipeline.utils import (
+    NamedDataPipelineLiterals as named_literals,
+    get_yaml_file_as_dict
+)
+from data_pipeline.etl_state import update_state, update_object_latest_dates
+from data_pipeline.utils.dags.airflow_s3_util_extension import (
+    S3NewKeyFromLastDataDownloadDateSensor,
+    S3HookNewFileMonitor
+)
+from data_pipeline.utils.dags.data_pipeline_dag_utils import (
+    get_default_args,
+    create_python_task
+)
+
+INITIAL_S3_XML_FILE_LAST_MODIFIED_DATE_ENV_NAME = (
+    "INITIAL_S3_XML_FILE_LAST_MODIFIED_DATE"
+)
+S3_BUCKET_POLLING_INTERVAL_IN_MINUTES_ENV_VAR_NAME = (
+    "S3_BUCKET_POLLING_INTERVAL_IN_MINUTES"
```

Comment:
I have a question regarding deployment. If I understand it correctly, we are merging all dags into one Airflow image. In that case the environment variable name may be too generic and should perhaps be prefixed by the project name? I could see us polling buckets at different intervals. For example the CSV bucket has a massive amount of files which takes a long time to poll. Whereas the XML bucket may still have a moderate amount of time, allowing us to poll less often. What do you think?

Reply:
this will require architectural rethink. in my opinion we have two options 

 - either we have a `controller dag pipeline` that has ability to manage polling frequencies at run time
 - or we deploy several pipelines with different polling frequencies

Reply:
I think I am just suggesting to add the a prefix like `EJP_XML_`. Maybe we could quickly discuss that in... 7 minutes.